Modify APoT dequantize method

asl3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82105
* __->__ #82126

### Summary
Modify APoT dequantize method to correctly add dequantized values to result numpy array and retain original tensor dimensions

### Test Plan
Run unit tests with: `python test/quantization/core/experimental/test_quantizer.py`

asl3 committed:
Modify APoT dequantize method

[ghstack-poisoned]

asl3 committed:
Update on "Modify APoT dequantize method"




[ghstack-poisoned]

HDCharles commented on torch/ao/quantization/experimental/quantizer.py:
```diff
@@ -57,15 +57,15 @@ def quantize(self, tensor2quantize: Tensor):
-        result = torch.from_numpy(result_temp).int()
+        result = torch.from_numpy(result_temp)
 
         return result
```
seems like the output of this for a 3x3 tensor would be a 1x9 tensor, shouldn't this output a tensor with the same output dimensions as the input tensor?

can you add a unit test for this?

asl3 replied:
yes agreed, fixed!

asl3 committed:
Update on "Modify APoT dequantize method"


### Summary
Modify APoT dequantize method to correctly add dequantized values to result numpy array

### Test Plan
Run unit tests with: `python test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

HDCharles commented on torch/ao/quantization/experimental/quantizer.py:
```diff
@@ -57,15 +57,20 @@ def quantize(self, tensor2quantize: Tensor):
+        result = torch.from_numpy(result_temp).reshape(orig_size)
 
-        result = torch.from_numpy(result_temp).int()
+        print(result)
```
this should be removed

HDCharles reviewed (approved):
there's a lint error and a print still in the code

otherwise looks good

asl3 committed:
Update on "Modify APoT dequantize method"


### Summary
Modify APoT dequantize method to correctly add dequantized values to result numpy array

### Test Plan
Run unit tests with: `python test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]
